app/controllers/doorkeeper/openid_connect/discovery_controller.rb

```diff
@@ -32,7 +32,7 @@ def provider_response
           introspection_endpoint: oauth_introspect_url(protocol: protocol),
           userinfo_endpoint: oauth_userinfo_url(protocol: protocol),
           jwks_uri: oauth_discovery_keys_url(protocol: protocol),
-          end_session_endpoint: openid_connect.end_session_endpoint.call,
+          end_session_endpoint: openid_connect.end_session_endpoint.call(self),
```

Comment:
I think this will crash if somebody defines a block in their initializer without arguments.

We could instead just use `instance_exec` and run the block in the scope of the controller:

```suggestion
          end_session_endpoint: instance_exec(&openid_connect.end_session_endpoint),
```

And then in the spec below we can just call `logout_url` directly.

Reply:
Well spotted, applied your suggestion now